src/proxy/topic/TopicProxy.ts

```diff
@@ -0,0 +1,53 @@
+import {BaseProxy} from "../BaseProxy";
+import {ITopic} from "../ITopic";
+import {MessageListener} from "../MessageListener";
+import {UUID} from "../../core";
+import {TopicAddMessageListenerCodec} from "../../codec/TopicAddMessageListenerCodec";
+import {TopicRemoveMessageListenerCodec} from "../../codec/TopicRemoveMessageListenerCodec";
+import {TopicPublishCodec} from "../../codec/TopicPublishCodec";
+import { ListenerMessageCodec} from "../../listener/ListenerMessageCodec";
+import {ClientMessage, ClientMessageHandler} from "../../protocol/ClientMessage";
+import {assertNotNull} from "../../util/Util";
+
+export class TopicProxy<E> extends BaseProxy implements ITopic<E> {
+
+    publish(message: E): Promise<void> {
+        assertNotNull(message);
+        const data = this.toData(message);
+        const request = TopicPublishCodec.encodeRequest(this.name, data);
+        const partitionId = request.getPartitionId();
+        return this.encodeInvokeOnPartition<void>(TopicAddMessageListenerCodec,  partitionId, TopicAddMessageListenerCodec.decodeResponse);
```

Comment:
After extending `PartitionSpecificProxy` you can just use `encodeInvoke` method of it.